CodeGeneration/Sources/generate-swift-syntax/templates/swiftsyntax/RawSyntaxNodesFile.swift

```diff
@@ -248,30 +315,42 @@ func rawSyntaxNodesFile(nodesStartingWith: [Character]) -> SourceFileSyntax {
               DeclSyntax(
                 """
                 let raw = RawSyntax.makeLayout(
-                  kind: .\(node.varOrCaseName), uninitializedCount: \(raw: node.children.count), arena: arena) { layout in
+                  kind: .\(node.varOrCaseName.callSiteMemberName), uninitializedCount: \(raw: node.children.count), arena: arena) { layout in
                   \(list)
                 }
                 """
               )
             } else {
-              DeclSyntax("let raw = RawSyntax.makeEmptyLayout(kind: .\(node.varOrCaseName), arena: arena)")
+              DeclSyntax("let raw = RawSyntax.makeEmptyLayout(kind: .\(node.varOrCaseName.callSiteMemberName), arena: arena)")
             }
             ExprSyntax("self.init(unchecked: raw)")
           }
 
           for (index, child) in node.children.enumerated() {
-            try VariableDeclSyntax(
-              "public var \(child.varOrCaseName.backtickedIfNeeded): Raw\(child.buildableType.buildable)"
-            ) {
-              let exclamationMark = child.isOptional ? "" : "!"
-
-              if child.syntaxNodeKind == .syntax {
-                ExprSyntax("layoutView.children[\(raw: index)]\(raw: exclamationMark)")
-              } else {
+            switch child.kind {
+            case .nodeChoices:
+              try VariableDeclSyntax(
+                "public var \(child.varOrCaseName.declarationSiteNameOrCallSiteVarName): \(child.isOptional ? TypeSyntax(OptionalTypeSyntax(wrappedType: child.syntaxChoicesType)) : child.syntaxChoicesType)"
```

Comment:
I think this can be slightly simplified.

```suggestion
                "public var \(child.varOrCaseName.declarationSiteNameOrCallSiteVarName): \(child.buildableType.optionalWrapped(child.syntaxChoicesType))"
```